### ISSUE TITLE
core dumped when inference run on webcam

### ISSUE THREAD

**Issue by kazmifactor** [cyrusbehr/YOLOv8-TensorRT-CPP#42]
i get this error when running the video file. can u help me with this. already tried installing libgtk2.0-dev and pkg-config

```
$ ./detect_object_video --model /home/inp/YOLOv8-TensorRT-CPP/models/yolov8n-pose.onnx --input 0
Searching for engine file with name: yolov8n-pose.engine.NVIDIAGraphicsDevice.fp16.1.1
Engine found, not regenerating...
Original video resolution: (640x480)
New video resolution: (1280x720)
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.9.0-dev) /home/inp/opencv/opencv/modules/highgui/src/window.cpp:1272: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvShowImage'

Aborted (core dumped)

```

**Comment by cyrusbehr** [comment]
After installing ibgtk2.0-dev and pkg-config, did you then re-compile OpenCV? You need to recompile it.

**Comment by kazmifactor** [comment]
> After installing ibgtk2.0-dev and pkg-config, did you then re-compile OpenCV? You need to recompile it.

thank you so much. it works. can we run it on a video and display the results?

**Comment by kazmifactor** [comment]
Hi i get this error 
![Uploading Screenshot from 2024-02-20 16-50-59.png…]()

**Comment by kazmifactor** [comment]
hi. i get this error.
![Screenshot from 2024-02-20 16-50-59](https://github.com/cyrusbehr/YOLOv8-TensorRT-CPP/assets/110693155/2f1652c3-db7a-44c7-a004-33c25fbd91e5)

**Comment by cyrusbehr** [comment]
Can you confirm the video file exists on disk.

What is the output of `file /home/inp/YOLOv8-TensorRT-CPP/images/test_vid.mp4`?

**Comment by kazmifactor** [comment]
> Can you confirm the video file exists on disk.
> 
> What is the output of running the following in your command line: `file /home/inp/YOLOv8-TensorRT-CPP/images/test_vid.mp4`? 

Yes the video file exists. I downloaded a video file from the internet. It opens a video

**Comment by cyrusbehr** [comment]
I just tested with the yolov8s.onnx model and this video and it works fine: `./detect_object_video --model ../models/yolov8s.onnx --input ~/Downloads/vid.mp4`

https://github.com/cyrusbehr/YOLOv8-TensorRT-CPP/assets/17056751/72697a7e-2f80-460b-b9cb-05e9f8f49fb6


https://github.com/cyrusbehr/YOLOv8-TensorRT-CPP/assets/17056751/1417542c-9b0f-40ca-bb84-12f4e5b8f8ca

**Comment by cyrusbehr** [comment]
Please make sure your video file is good. Test with the video I have provided.

**Comment by kazmifactor** [comment]
nope. same issue: the video still does not play. also you can check that i used `file /home/inp/YOLOv8-TensorRT-CPP/images/test_vid.mp4` to check what exists here
![Screenshot from 2024-02-21 09-40-25](https://github.com/cyrusbehr/YOLOv8-TensorRT-CPP/assets/110693155/cb7252d1-25d2-4247-bcbb-68cd99ca7714)

**Comment by cyrusbehr** [comment]
I'm not sure. It could be something wrong with your OpenCV installation. You should write a simple test program using OpenCV C++ and try to open the video stream and see if you can or not.

**Comment by cyrusbehr** [comment]
Ok. I'm not sure how much more support I can provide to be honest, short of sharing your screen and doing a session together. However for that level of support, I do charge.
